src/components/ResourceCard.js

```diff
@@ -412,16 +409,18 @@ export default function ResourceCard({
    * @todo Consider adding more validation for TSV properties as currently since it's quite generic.
    */
   const addRowToTsv = row => {
-    const { Reference: reference, ...rest } = row
+    const { Reference: reference } = row
     try {
       const { chapter: inputChapter, verse: inputVerse } =
-        tsvRowUtils.getChapterVerse(reference)
-        if (inputChapter !== chapter || inputVerse !== verse) {
-          // Todo: Do we then change the app's reference? Maybe yes
-          onTsvAdd(row, inputChapter, inputVerse, 0)
-          return
-        }
-        onTsvAdd(row, chapter, verse, itemIndex)
+        getChapterVerse(reference)
+
+      const newTsvs = (inputChapter !== Number(chapter) || inputVerse !== Number(verse))
+      // Todo: Do we then change the app's reference? Maybe yes
+        ? onTsvAdd(row, inputChapter, inputVerse, 0)
+        : onTsvAdd(row, chapter, verse, itemIndex)
+
```

Comment:
this is a bit dense using the trinary operator on these four lines.  Maybe break this out and explain what is happening here.